R/dash.R

```diff
@@ -637,12 +663,43 @@ Dash <- R6::R6Class(
     #' containing valid JavaScript, or a call to [clientsideFunction],
     #' including `namespace` and `function_name` arguments for a locally served
     #' JavaScript function.
+    #'
+    #'
+    #' For pattern-matching callbacks, the `id` field of a component is written
+    #' in JSON-like syntax which describes a dictionary object when serialized
+    #' for consumption by the Dash renderer. The fields include a `type` and
+    #' an `index`, which describe the targets of the callback.
```

Comment:
Does this mean we're only allowing `type` and `index` as keys (and in fact always using both of those keys)? That's much more restrictive than the Python version, where the fields can be any strings (except the empty string), and the values can be strings, numbers, and booleans, and should be serialized as such, ie no converting numbers or booleans to strings when these ids are serialized, and the keys should be sorted consistently. The id:
```
list("name" = "Alex", "age" = 43, "glasses" = TRUE)
```
should be stringified as:
```
'{"age":43,"glasses":true,"name":"Alex"}'
```
I don't actually know if the sorting matters in the end, but it might and we do enforce this in Python. Definitely flexible keys are important. We haven't made a lot of such examples because it quickly becomes a performance drag in the renderer and we're still working on that, but in principle you can have multiple wildcards, for example an output `list("row"=MATCH, "col"=MATCH, "type"="cell")` could depend on inputs `list("row"=MATCH, "type"="row header")` and `list("col"=MATCH, "type"="col header")`

Reply:
Looks like the code has been generalized to no longer require the keys to be `index` and `type`, which is great! Can we now generalize the comment above (or perhaps just remove that sentence) and also modify at least one test to use other keys?

Reply:
Added in 210d3c1